api/filter/http/gzip.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+
+// [#proto-status: experimental]
+
+package envoy.api.v2.filter.http;
+
+import "validate/validate.proto";
+
+// This is an HTTP filter which enables Envoy to compress dispatched data from an upstream
+// service upon client request. This is useful in situations where large payloads needs to
+// be transmitted without compromising the response time. Note that when this filter is
+// activated, content will be served with chunked transfer-encoding and gzip content-encoding.
+// TODO(gsagula): elaborate the last part in the final documentation.
+message Gzip {
+  // [#not-implemented-hide:] Value from 1 to 9 that controls the amount of internal memory
+  // used by zlib. Higher values usez more memory, but are faster and produce better compression
+  // results. Default value is 8.
+  uint32 memory_level = 1 [(validate.rules).uint32 = {gte: 1, lte: 9}];
```

Comment:
I think you will want to specify this (and content length below) as `optional` in the proto since there are default values if not specified. The wrapper types (`google.protobuf.UInt32Value`) are also inherently optional, so you can use that type.